File: RSA/Source.cpp
#include <iostream>

#include "Generator.h"
#include "Encryption.h"
#include "Decryption.h"
#include "Attack.h"

using namespace std;

int main() {
	pair<PublicKey, PrivateKey> keys = Generator::generateKey();

	cout << "Public key e = " << keys.first.e << endl;

	cout << "Private key d = " << keys.second.d << endl;

	cout << "Key N = " << keys.first.n << endl;


	cout << "Enter integer message to send to the other party" << endl;

	long long m;
	cin >> m;

	Encryption enc(keys.first);
	long long cEncryptedMessage = enc.Encrypt(m);

	Decryption dec(keys.second);
	Message mDecryptedMessage = dec.Decrypt(cEncryptedMessage);

	cout << "Sender message = " << m << endl;

	cout << "Encrypted message = " << cEncryptedMessage << endl;

	cout << "Decrypted message = " << mDecryptedMessage.message << endl;

	cout << (mDecryptedMessage.message == m ? "Encryption and Decryption succeeded" : "Failed to restore message") << endl;

	Message attacker_m = Attack::bruteforceAttack(keys.first, cEncryptedMessage);

	cout << "Attacker message = " << attacker_m.message << endl;

	cout << (attacker_m.message == m ? "Attacker succeeded to decrypt message" : "Attacker failed to decrypt message") << endl;
}

File: RSA - Python/Source.py
from Generator import Generator
from CommunicatingParty import CommunicatingParty
from PropertyOwner import PropertyOwner
from Attacker import Attacker


def main():
    keys = Generator.generate_key(300)
    print("n = " + str(keys[0].n))

    print("enter message to send")
    message = int(input())

    encrypter = CommunicatingParty(keys[0])
    encrypted_message = encrypter.encrypt(message)

    decrypter = PropertyOwner(keys[1])
    decrypted_message = decrypter.decrypt(encrypted_message)

    if message == decrypted_message:
        print("Encryption and Decryption succeeded")
    else:
        print("Encryption or Decryption failed")

    attacker_message = Attacker.chosen_cipher_attack(keys[0], encrypted_message, decrypter)
    if attacker_message == message:
        print("Chosen cipher attack succeeded")
    else:
        print("Chosen cipher attack failed")

    attacker_message = Attacker.brute_force_attack(keys[0], encrypted_message)
    if attacker_message == message:
        print("Brute force attack succeeded")
    else:
        print("Brute force attack failed")

if __name__ == '__main__':
    main()